### PR TITLE
SHOP-8: admin capability added

### PR DESCRIPTION
**Pull request by jacquelyn-bradshaw**

### CODE REVIEW

File: middlewares/authMiddleware.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   const isAuth = req.session.isAuthenticated
+  const isAdmin = req.session.isAdmin
 
   if (!user || !isAuth) {
     return next()
   }
 
   res.locals.isAuth = isAuth
 
+  if(!isAdmin) {
```
Comment (johnbradshw) [inline middlewares/authMiddleware.js]: Is this if block necessary?

File: views/includes/header.ejs [hunk 2]
```diff
@@ -5,8 +5,11 @@
       <% if (!locals.isAuth) { %>
       <li><a href="/signup">Signup</a></li>
       <li><a href="/login">Login</a></li>
-      <% } %> <% if (locals.isAuth) { %>
+      <% } else { %>
       <li><a href="/products">Products</a></li>
+      <% if (locals.isAdmin) { %>
```
Comment (johnbradshw) [inline views/includes/header.ejs:10]: You could do

```
if (!auth) {
   // do login
} else if (admin) {
   // do admin
} else {
   // do products 
}
```

**Review by johnbradshw — CHANGES_REQUESTED** [review]

File: middlewares/authMiddleware.js (hunk 1 above)
Comment (jacquelyn-bradshaw) [reply middlewares/authMiddleware.js]: If block not needed and removed

File: views/includes/header.ejs (hunk 2 above)
Comment (jacquelyn-bradshaw) [reply views/includes/header.ejs:10]: This does not allow admin to view the products page so not changed

**Review by johnbradshw — APPROVED** [review]

**Pull request merged**